pkg/handlers/generic/lifecycle/utils/utils.go

```diff
@@ -153,40 +144,6 @@ func RetrieveValuesTemplateConfigMap(
 	return configMap, nil
 }
 
-func CreateStorageClass(
-	storageConfig v1alpha1.StorageClassConfig,
-	defaultsNamespace string,
-	provisionerName v1alpha1.StorageProvisioner,
-	isDefault bool,
-) *storagev1.StorageClass {
-	var params map[string]string
-	if provisionerName == v1alpha1.AWSEBSProvisioner {
-		params = defaultAWSStorageClassParams
-	}
-	if storageConfig.Parameters != nil {
-		params = maps.Clone(storageConfig.Parameters)
-	}
```

Comment:
This dropped all defaults if a user set any parameters. In the new implementation, the defaults are set first, and then user provided parameters are added, overriding any default values that have the same key.
This gives us the flexibility to set sane defaults, but still give the end user control to change it if needed.